lib/phoenix/controller.ex

```diff
@@ -128,6 +128,12 @@ defmodule Phoenix.Controller do
   def endpoint_module(conn), do: conn.private.phoenix_endpoint
 
   @doc """
+  Returns the socket handler module as an atom, raises if unavailable.
+  """
+  @spec socket_handler_module(Plug.Conn.t) :: atom
+  def socket_handler_module(conn), do: conn.private.phoenix_socket_handler
```

Comment:
Why don't you pass it as argument to the transport then? :) They don't need to mimic plug APIs.